### DISCUSSION TITLE
Redis example needs pool not a client

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/tower-sessions#139] [Q&A] [answered]
`let session_store = RedisStore::new(pool);`

this line gives err
because it needs a client not a pool..

while I prefer pool not client .. as the example suggests .. but it is just gives err

```
   |
26 |     let session_store = RedisStore::new(pool);
   |                         --------------- ^^^^ expected `RedisClient`, found `RedisPool`
   |                         |
   |                         arguments to this function are incorrect
   ```

**Comment by maxcountryman** [comment]
Here `client` is just a naming convention, but [its type](https://github.com/maxcountryman/tower-sessions/blob/253df1eb1fe34bd297fca57cd13e9fe26f453cb2/redis-store/src/lib.rs#L40C9-L40C22) allows for both a pool and a client.

The [example](https://github.com/maxcountryman/tower-sessions/blob/253df1eb1fe34bd297fca57cd13e9fe26f453cb2/examples/redis-store.rs#L15-L20) illustrates using a pool.

**Reply by HosMercury** [reply]
```
    let pool = RedisPool::new(RedisConfig::default(), None, None, None, 6).unwrap();

    let redis_conn = pool.connect();
    pool.wait_for_connect().await.unwrap();

    let session_store = RedisStore::new(pool);
    let session_layer = SessionManagerLayer::new(session_store)
        .with_secure(false)
        .with_expiry(Expiry::OnInactivity(Duration::seconds(10)));
        ```
        
        
i have the err above although I copied the code 
`tower-sessions = { version = "0.7.0", features = ["redis-store"] }`

**Comment by HosMercury** [comment, marked as answer]
it was conflicting versions
